x-pack/plugins/apm/server/lib/alerts/register_transaction_duration_alert_type.ts

```diff
@@ -47,116 +43,124 @@ const paramsSchema = schema.object({
+}: RegisterRuleDependencies) {
+  registry.registerType(
+    createAPMLifecyleRuleType({
+      id: AlertType.TransactionDuration,
+      name: alertTypeConfig.name,
+      actionGroups: alertTypeConfig.actionGroups,
+      defaultActionGroupId: alertTypeConfig.defaultActionGroupId,
+      validate: {
+        params: paramsSchema,
+      },
+      actionVariables: {
+        context: [
+          apmActionVariables.serviceName,
+          apmActionVariables.transactionType,
+          apmActionVariables.environment,
+          apmActionVariables.threshold,
+          apmActionVariables.triggerValue,
+          apmActionVariables.interval,
+        ],
+      },
+      producer: 'apm',
+      minimumLicenseRequired: 'basic',
+      executor: async ({ services, params }) => {
+        const config = await config$.pipe(take(1)).toPromise();
+        const alertParams = params;
+        const indices = await getApmIndices({
+          config,
+          savedObjectsClient: services.savedObjectsClient,
+        });
 
-      const searchParams = {
-        index: indices['apm_oss.transactionIndices'],
-        size: 0,
-        body: {
-          query: {
-            bool: {
-              filter: [
-                {
-                  range: {
-                    '@timestamp': {
-                      gte: `now-${alertParams.windowSize}${alertParams.windowUnit}`,
+        const searchParams = {
+          index: indices['apm_oss.transactionIndices'],
+          size: 0,
+          body: {
+            query: {
+              bool: {
+                filter: [
+                  {
+                    range: {
+                      '@timestamp': {
+                        gte: `now-${alertParams.windowSize}${alertParams.windowUnit}`,
+                      },
                     },
                   },
-                },
-                { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
-                { term: { [SERVICE_NAME]: alertParams.serviceName } },
-                { term: { [TRANSACTION_TYPE]: alertParams.transactionType } },
-                ...environmentQuery(alertParams.environment),
-              ],
+                  { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
+                  { term: { [SERVICE_NAME]: alertParams.serviceName } },
+                  { term: { [TRANSACTION_TYPE]: alertParams.transactionType } },
+                  ...environmentQuery(alertParams.environment),
+                ] as QueryContainer[],
+              },
             },
-          },
-          aggs: {
-            agg:
-              alertParams.aggregationType === 'avg'
-                ? { avg: { field: TRANSACTION_DURATION } }
-                : {
-                    percentiles: {
-                      field: TRANSACTION_DURATION,
-                      percents: [
-                        alertParams.aggregationType === '95th' ? 95 : 99,
-                      ],
+            aggs: {
+              metric:
```

Comment:
https://github.com/elastic/kibana/blob/d678ffb332015f9a35c16df8ac90c54aa6342186/x-pack/plugins/apm/server/lib/helpers/latency_aggregation_type/index.ts#L10-L28 does this but it calls it `latency` instead of `metric`.